server/src/main/java/org/opensearch/index/search/stats/SearchStats.java

```diff
@@ -61,6 +62,31 @@ public class SearchStats implements Writeable, ToXContentFragment {
      *
      * @opensearch.internal
      */
+
+    public static class RequestStatsLongHolder {
```

Comment:
This class will need to be modified (and serialization will need to change) if we add new search phases in future.

It's safer to serialize it as a list as follows:
1. Emit an integer signifying how many phases have stats. If the stats holder is `null` or empty, then emit 0.
2. For each phase, emit the phase name as a string, followed by the three long values.

Reply:
I don't think we need to do any over-engineering here by using list and all those things. We rarely add new search phases, even if needed it is just addition of couple of lines. It is done similarly for shard related stats.

Reply:
This was already taken care of as discussed elasewhere.